mern-ecommerce/controllers/order.js

```diff
@@ -61,3 +61,15 @@ exports.updateOrderStatus = (req, res) => {
     }
   );
 };
+
+exports.deleteOrder = async (req, res) => {
+  try {
+    const order = await Order.findByIdAndRemove(req.params.orderId);
+    if (order) {
+      return res.status(404).json({ message: "Order does not exist" });
+    }
+    res.status(200).json({ message: "Order deleted successfully" });
+  } catch (err) {
+    res.status(500).json({ error: errorHandler(err) });
```

Comment:
The error handling mechanism is correctly implemented, but it would be better to include a more descriptive error message. Instead of just sending `{ error: errorHandler(err) }`, consider sending `{ error: 'Internal Server Error', message: errorHandler(err) }`. This will provide more information to the client about the error that occurred.